src/index.js

```diff
@@ -8,272 +8,358 @@ const _ = require("lodash");
+        "expected-output"
+      )
+    ) {
+      continue;
+    }
 
-    core.info("Comparing api output with expected output...");
-    for (let i = 0; i < successResults.length; i++) {
-      let transformerVersionID = successResults[i].transformerVersionID;
-      if (!transformationDict.hasOwnProperty(transformerVersionID)) {
-        continue;
-      }
+    const expectedOutputfile =
+      transformationDict[transformerVersionID]["expected-output"];
+    const expectedOutput = expectedOutputfile
+      ? JSON.parse(fs.readFileSync(expectedOutputfile))
+      : "";
 
-      const apiOutput = successResults[i].result.output.transformedEvents;
-      const transformationName = transformationDict[transformerVersionID].name;
-      const transformationHandleName = _.camelCase(transformationName);
+    if (expectedOutput == "") {
+      continue;
+    }
+
+    if (!isEqual(expectedOutput, actualOutput)) {
+      core.info(
+        `Test output do not match for transformation: ${transformationName}`
+      );
+      outputMismatchResults.push(
+        `Test output do not match for transformation: ${transformationName}`
+      );
 
       fs.writeFileSync(
-        `${testOutputDir}/${transformationHandleName}_output.json`,
-        JSON.stringify(apiOutput, null, 2)
+        `${testOutputDir}/${transformationHandleName}_diff.json`,
+        JSON.stringify(detailedDiff(expectedOutput, actualOutput), null, 2)
       );
+
       testOutputFiles.push(
-        `${testOutputDir}/${transformationHandleName}_output.json`
+        `${testOutputDir}/${transformationHandleName}_diff.json`
       );
-
-      if (
-        !transformationDict[transformerVersionID].hasOwnProperty(
-          "expected-output"
-        )
-      ) {
-        continue;
-      }
-
-      let expectedOutputfile =
-        transformationDict[transformerVersionID]["expected-output"];
-      let expectedOutput = expectedOutputfile
-        ? JSON.parse(fs.readFileSync(expectedOutputfile))
-        : "";
-
-      if (expectedOutput == "") {
-        continue;
-      }
-
-      if (!isEqual(expectedOutput, apiOutput)) {
-        core.info(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-        outputMismatchResults.push(
-          `Test output do not match for transformation: ${transformationName}`
-        );
-
-        fs.writeFileSync(
-          `${testOutputDir}/${transformationHandleName}_diff.json`,
-          JSON.stringify(detailedDiff(expectedOutput, apiOutput), null, 2)
-        );
-
-        testOutputFiles.push(
-          `${testOutputDir}/${transformationHandleName}_diff.json`
-        );
-      }
     }
+  }
+  return { outputMismatchResults, testOutputFiles };
+}
+
+// Upload the test results to an artifact store.
+async function uploadTestArtifacts(testOutputFiles) {
+  core.info(`Uploading test artifacts`);
+  // upload artifact
+  if (uploadTestArtifact) {
```

Comment:
Can we move this check layer above as well so this function just simply uploadTestArtifacts only. Wdyt ?